agnssDeviceRequestsHandler/agnssDeviceRequestsHandler.ts

```diff
@@ -63,17 +63,21 @@ const agnssDeviceRequestsHandler: AzureFunction = async (
 
 	// Find A-GNSS requests
 	const agnssRequests = requests
-		.map((request, i) => ({
-			request,
-			deviceId:
-				context.bindingData.systemPropertiesArray[i][
-					'iothub-connection-device-id'
-				],
-			properties: context.bindingData.propertiesArray[i] as Record<
-				string,
-				string
-			>,
-		}))
+		.map((request, i) => {
+			const systemPropertiesArray = Array.isArray(
+				context.triggerMetadata?.systemPropertiesArray,
```

Comment:
Why is the check on the array property needed whether it is an array?

Reply:
The `context` is `InvocationContext` type where `triggerMetadata` has `Record<string, unknown>` type. Therefore, `systemPropertiesArray` is unknown type. 

I think assertion here is needed.

Reply:
But we know this property is an array, it IS in the name: `systemPropertiesArray`. So the type needs to be corrected.